pulp_container/app/migrations/0038_add_manifest_metadata_fields.py

```diff
@@ -0,0 +1,148 @@
+# Generated by Django 4.2.10 on 2024-02-29 16:04
+
+import json
+
+from contextlib import suppress
+
+from django.db import migrations, models
+from django.core.paginator import Paginator
+from django.core.exceptions import ObjectDoesNotExist
+
+from pulp_container.constants import MEDIA_TYPE
+
+PAGE_CHUNK_SIZE = 200
+
+
+def populate_annotations_and_labels(apps, schema_editor):
+    Manifest = apps.get_model('container', 'Manifest')
+
+    manifests = Manifest.objects.exclude(
+        media_type__in=[MEDIA_TYPE.MANIFEST_LIST, MEDIA_TYPE.INDEX_OCI]
+    ).order_by('pulp_id')
+    update_manifests(manifests)
+
+    manifest_lists = Manifest.objects.filter(
+        media_type__in=[MEDIA_TYPE.MANIFEST_LIST, MEDIA_TYPE.INDEX_OCI]
+    ).order_by('pulp_id')
+    update_manifests(manifest_lists)
+
+
+def update_manifests(manifests_qs):
+    paginator = Paginator(manifests_qs, PAGE_CHUNK_SIZE)
+    for page_num in paginator.page_range:
+
+        manifests_to_update = []
+        page = paginator.page(page_num)
+        for obj in page.object_list:
+            if m := update_manifest(obj):
+                manifests_to_update.append(m)
+
+        if manifests_to_update:
+            fields_to_update = ['annotations', 'labels', 'is_bootable', 'is_flatpak']
+            manifests_qs.model.objects.bulk_update(
+                manifests_to_update,
+                fields_to_update,
+            )
+
+
+def update_manifest(manifest):
+    with suppress(ObjectDoesNotExist):
+        artifact = manifest._artifacts.get()
```

Comment:
i don't know if we should keep this as a migration or rather a management command. this is expensive

Reply:
Makes sense. Let me write a new `populate-annotations-labels` command.

Reply:
see 0033_raise_warning_for_repair.py